### DISCUSSION TITLE
Pagination support for api/events API endpoint?

### DISCUSSION THREAD

**Discussion by chrisharrington** [blakeblackshear/frigate#5593] [Ask A Question] [answered]
Hi there!

I'm making use of the Frigate HTTP API, specifically the `api/events` endpoint. I see that there are a number of query string parameters around filtering and limiting the response, but I don't see anything that would allow me to implement pagination. In my application, I'd like to implement a table with pagination. Is there a way to do that using the current API implementation?

Thanks!
Chris

**Comment by blakeblackshear** [comment, marked as answer]
You can use the start and end times to do something similar to paging. That's how the events page in the UI works.

**Reply by chrisharrington** [reply]
Ah yeah, I see what's going on. Take 25 rows, then the next "page" will be the following 25 based on `end_time` using the `before` filter. That'll work. Thanks!
